cli/src/semgrep/rule_match.py

```diff
@@ -18,16 +18,20 @@
 from attrs import evolve
 from attrs import field
 from attrs import frozen
+from boltons.cacheutils import cachedmethod
```

Comment:
Can you comment on the motivation for the line cache? Mostly just because caching is natoriously hard to get right.